time/src/time.cpp

```diff
@@ -0,0 +1,47 @@
+#include "lute/time.h"
+#include "uv.h"
+
+#include <iterator>
+#include <assert.h>
+
+namespace libtime
+{
+int lua_now(lua_State* L)
+{
+    uv_timespec64_t timespec;
+
+    int status = uv_clock_gettime(UV_CLOCK_MONOTONIC, &timespec);
+    assert(status == 0);
+
+    // both of these are ints, so we cast them to double.
+    lua_pushnumber(L, static_cast<double>(timespec.tv_sec) + static_cast<double>(timespec.tv_nsec / 1e9));
```

Comment:
We should not return an instant as a number to luau code. The goal here is to provide a userdata for the timespec instead, and the necessary functionality to support it interacting with durations appropriately.

Reply:
It is now a userdata